_posts/2017-09-25-scastie-embedding.md

```diff
@@ -0,0 +1,140 @@
+---
+category: blog
+by: Guillaume Massé
+title: "Announcing Scastie Embedding"
+date: 25-9-2017
+---
+
+We are excited to announce the availability of a new core feature of Scastie: Embedding.
+
+# What is Scastie?
+
+If you don't know about Scastie yet, here's a recap from our previous announcement.
+
+**Scastie is... Scala + sbt in your browser!** You can:
+
+- use any version of Scala (scalac), or even alternate backends such as Dotty, Scala.js, Scala Native, and Typelevel Scala.
+- use any library published on Maven.
+- save and share Scala programs/builds with anybody
+
+This is all possible because Scastie is based on sbt, allowing us to support newer
+Scala versions and resolve library dependencies on Maven.
+
+Want to see what Scastie can do? Check out this embedding below!
+
+<script src="https://scastie.scala-lang.org/d4Zr5VLvQAirvPwIr2bqMA.js"></script>
+
+## How do I embed Scastie in my website/documentation
+
+Since it's possible to use external libraries, Scastie is an excellant tool to add to your library documentation or your blog post. Let's explore the available methods to bootstrap the embedding mode.
+
+You can embed a snippet with two different ways. You can embed a resource that
+was previously saved or you can convert code to a snippet.
+
+### Embedded Resources
+
+#### Via HTML
+
+From the Scastie website, once you save a snippet you will have a button that opens some html code you can paste into your document.
+
+<img alt="scastie screenshot" src="/resources/img/blog/scastie-embedded/embedded-button.gif">
+
+Any saved resource can be embedded via a script tag by appending ".js" to its URL.
+
+```html
+
+<!-- Annonymous Resource -->
+<script src="https://scastie.scala-lang.org/xbrvky6fTjysG32zK6kzRQ.js"></script>
+
+<!-- User's Resource -->
+<script src="https://scastie.scala-lang.org/MasseGuillaume/CpO2s8v2Q1qGdO3vROYjfg.js"></script>
+<script src="https://scastie.scala-lang.org/MasseGuillaume/CpO2s8v2Q1qGdO3vROYjfg/1.js"></script>
+```
+
+The embedding will be patched at the location of the script tag.
+
+It's also possible to adjust the theme via the GET parameter theme. For example:
+
+```html
+<script src="https://scastie.scala-lang.org/xbrvky6fTjysG32zK6kzRQ.js?theme=dark"></script>
+```
+
+The theme can be `dark` or `light`. The default value is `light`.
+
+Finally, it's also possible to customize the css such that it blends into your website.
+
+Take a look at: [https://codepen.io/MasseGuillaume/pen/oeKoee](https://codepen.io/MasseGuillaume/pen/oeKoee)
+
+#### Via Javascript
+
+```html
+
+<!-- Embedding Script -->
+<div id="resource"></div>
+<script src="https://scastie.scala-lang.org/embedded.js"></script>
+
+<script>
+window.addEventListener('load', function() {
+  scastie.Embedded('#resource-1', { 
```

Comment:
Is it expected that the selector doesn’t match the id of the div above?

Reply:
ops, fixed